astropy/cosmology/core.py

```diff
@@ -165,22 +353,98 @@ def __eq__(self, other):
         Returns
         -------
         bool
-            True if all immutable fields are equal, False otherwise.
+            True if cosmologies are equivalent, False otherwise.
         """
+        # The options are: 1) same class & parameters; 2) same class, different
```

Comment:
How does Numpy do ``1 == array`` if python only "call[s] ``other.__eq__(self)`` if other is a strict subclass of self" ?

Reply:
Sorry, I was unclear: if the two are different classes, both get called; but if `isinstance(other, type(self)) and type(other) is not type(self)`, then only `other.__eq__(self)` is called - since presumably, the subclass knows how to deal with the superclass. See  https://docs.python.org/3/reference/datamodel.html#object.__radd__

(Actually, that suggests `self.__eq__(other)` may still get called, but just second - not completely sure!!)

Reply:
Thanks for the explanation! I'll make a card for followup on this. Seems good to have it implemented exactly like python, but lower priority because it already works as expected.

Reply:
https://github.com/astropy/astropy/projects/7#card-69390093